PalCalc.Solver/BreedingSolver.cs

```diff
@@ -25,13 +33,23 @@ public class SolverStatus
         public int CurrentStepIndex { get; set; }
         public int TargetSteps { get; set; }
         public bool Canceled { get; set; }
+        public bool Paused { get; set; }
 
         public int CurrentWorkSize { get; set; }
 
         public int WorkProcessedCount { get; set; }
         public int TotalWorkProcessedCount { get; set; }
     }
 
+    public class SolverStateController
+    {
+        public CancellationToken CancellationToken { get; set; }
+        public bool IsPaused { get; private set; }
+
+        public void Pause() => IsPaused = true;
+        public void Resume() => IsPaused = false;
+    }
```

Comment:
Would be better to have the solver return a `SolverJob` which had these controls

Reply:
Going to punt on this for now